### DISCUSSION TITLE
Windows CMD: Escape special characters in variable values

### DISCUSSION THREAD

**Discussion by Taitava** [Taitava/obsidian-shellcommands#106] [Ideas]
I'm adding this just as a quick note on something I need to test and research more later. This would be an improvement for some later version, this will not come to `0.7.0`.

So far I haven't been aware of an ability to escape special characters in Windows' CMD shell. But I read very quickly that maybe `^` could be used as an escape character: https://ss64.com/nt/syntax-esc.html

I need to test that can I use `^` in front of every character that SC considers as a _special character_.

**Comment by FelipeRearden** [comment]
> I need to test that can I use `^` in front of every character that SC considers as a _special character_.

Forgive for my comment, I don't if it is related to this issue: 

Just to remember that Obsidian uses `^` for block refs `[[note_name#^block_id]]`

Have a great day!

**Reply by Taitava** [reply]
It's good to bring up this kind of observations 👍 . This one should not be a problem. If a variable value happens to contain a `^` character (e.g. in a link like you meantioned), **I guess** the character will be escaped by making it to be `^^`. The first `^` tells that the second `^` should be used literally, without any special meaning. But the future will show me how escaping in CMD really works, and do I have the correct idea about it or not.

So, to summarize: I think `^` in Obsidian notes will be taken care of just fine. 🙂

**Reply by FelipeRearden** [reply]
> It's good to bring up this kind of observations 👍 .

Thanks @Taitava ! It's important as a tester/user to bring any concept about Obsidian that I know :)
